runtime/compiler/spotbugs.vim

```diff
@@ -0,0 +1,21 @@
+" Vim compiler file
+" Compiler: Spotbugs (Java static checker; needs javac compiled classes)
+" Maintainer:   @Konfekt
+" Last Change:  2024 nov 6
+
+if exists("current_compiler") | finish | endif
+let current_compiler = "spotbugs"
+
+let s:cpo_save = &cpo
+set cpo&vim
+
+" Unfortunately Spotbugs does not output absolute paths, so you need to
+" pass the directory of the files being checked as `-sourcepath` parameter.
+" See https://github.com/spotbugs/spotbugs/issues/856
+exe 'CompilerSet makeprg=spotbugs'..(has('win32')?'.bat':'')..
+  \ escape(' -textui -emacs -effort:max -sourcepath %:p:h:S %:p:h:S', ' ')
```

Comment:
Is that correct? Shouldn't that be rather:
```
exe 'CompilerSet makeprg=spotbugs'..(has('win32')?'.bat':'')..
  \ escape(' -textui -emacs -effort:max -sourcepath %:p:h:S %:p:S', ' ')
```
so the directory of the current file for the `-sourcepath` argument and then the full path of the file currently being edited?

Reply:
It's been on purpose. Spotbugs lints class files and requires all dependencies as input. So linting a single one, `%:p:S:r.class` would lead to complaints about missing dependent class files as input. Listing the exact dependent class files as input seems unpractical.

However, now that you asked about, I dug deeper and seemingly found a way to only analyze the currently open buffer and its dependent classes